python/cudf/cudf/core/column/datetime.py

```diff
@@ -141,7 +141,8 @@ def as_numerical_column(self, dtype, **kwargs):
 
     def as_string_column(self, dtype, **kwargs):
         from cudf.core.column import string
-
+        if not kwargs.get("format"):
+            kwargs["format"] = "%Y-%m-%d %H:%M:%S.%f"
```

Comment:
Since we can't pass `format` to astype anymore, we need to pick a format at some point that we're going to dump the datetimes into when we cast to string. The cython layer currently provides a default format but it doesn't seem to line up with what pandas does. Pandas stores everything as nanoseconds internally, but even though there seems to be one single storage format there's apparently multiple output formats for a string cast:

```
>>> pd.Series(['2011-01-01 09:30:00.031200'], dtype='datetime64[ns]').astype('str')
0    2011-01-01 09:30:00.031200
dtype: object
>>> pd.Series(['2011-01-01'], dtype='datetime64[ns]').astype('str')
0    2011-01-01
dtype: object
```

I'm still looking into exactly how it decides what format to use in the output but based on what I'm finding so far it appears the data is scanned and rounded if possible, I'm not sure if we do/should do this.

Reply:
Given that we support more than just `ns` in cuDF, would it make sense to just have a lookup table of formats --> dtype? i.e. if we have `s` we don't need the `ns` format.

Reply:
Yeah, that's what I think we should do as well. Will update this shortly.